File: ToDoApp/Frontend todoapp/myapp/src/Services/TaskService.js
// Import axios library for making HTTP requests
import axios from "axios";

// Define base URL for REST API
const REST_API_BASE_URL = "http://localhost:8080/todoapp/tasks";

// Function to fetch all tasks from the server
export const listTasks = () => axios.get(REST_API_BASE_URL);

// Function to create a new task on the server
export const createTask = (task) => axios.post(REST_API_BASE_URL, task);

// Function to fetch a specific task from the server by its id
export const getTask = (taskId) => axios.get(REST_API_BASE_URL + "/" + taskId);

// Function to update a task on the server with a specific id
export const updateTask = (taskId, task) =>
  axios.put(REST_API_BASE_URL + "/" + taskId, task);

// Function to delete a task from the server by its id
export const deleteTask = (taskId) =>
  axios.delete(REST_API_BASE_URL + "/" + taskId);

File: ToDoApp/ToDo/src/main/java/com/todoapp/service/TaskService.java
package com.todoapp.service;

//Define a service interface for managing tasks


import java.util.List;

import com.todoapp.dto.TaskDto;

//Interface TaskService outlines methods for task management
public interface TaskService {
	// Method to create a new task
	TaskDto createTask(TaskDto taskDto);

	// Method to retrieve a task by its id
	TaskDto getTaskById(Long taskId);

	// Method to retrieve all tasks
	List<TaskDto> getAllTasks();

	// Method to update a task
	TaskDto updateTask(Long taskId, TaskDto updatedTask);

	// Method to delete a task
	void deleteTask(Long taskId);
}